### PR TITLE
MAINT: Prefer signature over formatargspec

### PR DESCRIPTION
**Pull request by bashtage**
Prefer modern method to get signature over deprecated method

### CODE REVIEW

File: statsmodels/base/wrapper.py [hunk 1]
```diff
@@ -96,10 +96,15 @@ def wrapper(self, *args, **kwargs):
...
+    try:  # Python 3.3+
+        sig = inspect.signature(func)
+        formatted = str(sig)
+    except AttributeError:
+        # TODO: Remove when Python 2.7 is dropped
+        argspec = getargspec(func)
+        formatted = inspect.formatargspec(argspec[0],
+                                          varargs=argspec[1],
+                                          defaults=argspec[3])
     func_name = get_function_name(func)
```
Comment (jbrockmendel) [inline statsmodels/base/wrapper.py]: OT: I'm pretty sure in all supported versions we can just use `func.__name__` at this point

**Comment by jbrockmendel** [comment]
LGTM

File: statsmodels/base/wrapper.py (hunk 1 above)
Comment (bashtage) [reply statsmodels/base/wrapper.py]: Seems to be the case

**Comment by josef-pkt** [comment]
LGTM, merging

@bashtage @jbrockmendel Thank you

**Comment by josef-pkt** [comment]
do you know why master fails?
I don't see where there is a fail.

https://travis-ci.org/statsmodels/statsmodels/builds/435666459

**Comment by bashtage** [comment]
I think there is a bug in travis that is making it fail when programs try and write too much to the console too quickly.  It only started in the past couple of days.  Silencing warnings from pytest will let it pass.

**Comment by jbrockmendel** [comment]
> Silencing warnings from pytest will let it pass.

booyah!  #5260, #5204

**Comment by bashtage** [comment]
It only appears with coverage since coverage also writes a lot of content (from what I can tell).

**Pull request merged**